CMakeLists.txt

```diff
@@ -0,0 +1,202 @@
+# Written in 2016-2017 by Henrik Steffen Gaßmann henrik@gassmann.onl
+#
+# To the extent possible under law, the author(s) have dedicated all
+# copyright and related and neighboring rights to this software to the
+# public domain worldwide. This software is distributed without any warranty.
+#
+# You should have received a copy of the CC0 Public Domain Dedication
+# along with this software. If not, see
+#
+#     http://creativecommons.org/publicdomain/zero/1.0/
+#
+########################################################################
+cmake_minimum_required(VERSION 3.1)
+
+project(base64 LANGUAGES C VERSION 0.3.0)
+
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules")
+
+#######################################################################
+# platform detection
+include(TargetArch)
+detect_target_architecture(_TARGET_ARCH)
+
+
+########################################################################
+# Compilation options
+option(BASE64_INSTALL_TARGET "add an install target" ON)
+option(BASE64_BUILD_TESTS "add test projects" ON)
+option(BASE64_WITH_OPENMP "use openmp" OFF)
+
+if (_TARGET_ARCH STREQUAL "x86" OR _TARGET_ARCH STREQUAL "x64")
+    option(BASE64_WITH_FAST_UNALIGNED_ACCESS "vectorization: unaligned access enabled" ON)
+    option(BASE64_WITH_SSSE3 "add SSSE 3 codepath" ON)
+    option(BASE64_WITH_SSE41 "add SSE 4.1 codepath" ON)
+    option(BASE64_WITH_SSE42 "add SSE 4.2 codepath" ON)
+    option(BASE64_WITH_AVX "add AVX codepath" ON)
+    option(BASE64_WITH_AVX2 "add AVX 2 codepath" ON)
+
+elseif (_TARGET_ARCH STREQUAL "arm")
+    option(BASE64_WITH_NEON32 "add NEON32 codepath" OFF)
+
+elseif (_TARGET_ARCH STREQUAL "arm64")
+    option(BASE64_WITH_NEON64 "add NEON64 codepath" ON)
+
+endif()
+
+###################################################################
+# OpenMP
+if(BASE64_WITH_OPENMP)
+    find_package(OpenMP)
+    if (OPENMP_FOUND)
+        option(BASE64_USE_OpenMP "Utilize OpenMP to parallelize encoding and decoding." ON)
+    endif()
+endif()
+
+########################################################################
+# library project
+add_library(base64 STATIC
```

Comment:
See https://github.com/aklomp/base64/issues/70

I think shared libraries are possible.

Reply:
This ⬆️ .
There is no need to emulate autotools behavior with building shared and static libraries simultaneously.
Users will use either static or shared libraries.